### PR TITLE
Sub branch

### PR DESCRIPTION
**Pull request by vikasreddy-elixr**
This is the code for the Java terminal application where it takes two command line args from the user and checks for the file through path given and search for the word and word count.

### CODE REVIEW

File: src/com/company/Main.java [hunk 1]
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
 
         String txtfile = args[0];
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Please check what will happen if args[0] does not contain anything. For example if user did not pass any argument what happens to the program?

File: src/com/company/Main.java [hunk 2]
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
...
         File file = new File(txtfile);
-        checkforsupport(file);
-        checkforexistance(file);
+        checkForSupport(file);
+        checkForExistance(file);
         String searchword = args[1];
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Please check what will happen if args[1] does not contain anything. For example if user did not pass any argument what happens to the program?

File: src/com/company/Main.java [hunk 3]
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
...
         System.out.println("processing................");
         String data = null;
         data = readFileAsString(txtfile);
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Cant we combine line number 19 and 20?

File: src/com/company/Main.java (hunk 1 above)
Comment (kssreejith1994) [inline src/com/company/Main.java]: txtfile should be renamed in camel case. Also wouldn't it be better if we rename the txtfile to something like txtFilePath or so?

File: src/com/company/Main.java (hunk 3 above)
Comment (kssreejith1994) [inline src/com/company/Main.java]: What if the file is empty? If we do a check there, we can stop the execution then and there and no need to proceed further.

File: src/com/company/Main.java [hunk 4]
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
 
         String txtfile = args[0];
         File file = new File(txtfile);
-        checkforsupport(file);
-        checkforexistance(file);
+        checkForSupport(file);
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: This method is a void method and if the file does not exists also it just prints "file does not exists" and program execution continues right? You might want to take care of this? Same with checkForSupport  and checkForExistence too

File: src/com/company/Main.java [hunk 5]
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
...
+        checkForExistance(file);
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Might be good if we change the method name lie checkIfFileExists(arguments)

File: src/com/company/Main.java [hunk 6]
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
...
         String searchword = args[1];
         System.out.println("processing................");
         String data = null;
         data = readFileAsString(txtfile);
-        avoidspecialchar(data);
-        searchtheword(data,searchword);
+        avoidSpecialChar(data);
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Please rename to removeSpecialCharacters

**Review by kssreejith1994 — CHANGES_REQUESTED** [review]
Good work, but please make the changes suggested.

File: src/com/company/Main.java (hunk 1 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: it display the file not exists message

File: src/com/company/Main.java (hunk 3 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: yeah we can do that.
resolved it.

File: src/com/company/Main.java (hunk 1 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: yeah!!
done with that..

File: src/com/company/Main.java [hunk 7]
```diff
@@ -36,7 +36,7 @@ public static String readFileAsString(String fileName)
               return data;
     }
 
-        public static void checkforsupport(File file)
+        public static void checkForSupport(File file)
```
Comment (premsrj) [inline src/com/company/Main.java]: You should ideally be throwing an exception from here or atleast return false if the file is not in a supported format

File: src/com/company/Main.java [hunk 8]
```diff
@@ -50,7 +50,7 @@ public static void checkforsupport(File file)
                 }
         }
 
-        public static void checkforexistance(File file) {
+        public static void checkForExistance(File file) {
```
Comment (premsrj) [inline src/com/company/Main.java]: You should ideally be throwing an exception from here or atleast return false if the file is not in a supported format

File: src/com/company/Main.java [hunk 9]
```diff
@@ -59,10 +59,11 @@ public static void checkforexistance(File file) {
                 System.out.println("file doesn't found ");
             }
         }
-        public static void avoidspecialchar(String data) {
+        public static void avoidSpecialChar(String data)
+        {
             data = data.replaceAll("[^a-zA-Z0-9@]", " ");
```
Comment (premsrj) [inline src/com/company/Main.java]: It is always better to return the modified data, instead of actually modifying the reference inside the method.

File: src/com/company/Main.java [hunk 10]
```diff
@@ -8,18 +8,40 @@
...
-    public static void main(String[] args){
+    public static void main(String[] args) {
+
+        String txtFilePath = args[0];
+
+        File file = new File(txtFilePath);
+        if(file.length()==0)
+        {
+            System.out.println("Enter the valid path ");
+            System.exit(0);
```
Comment (premsrj) [inline src/com/company/Main.java]: Why not just return from this method instead of explicitly calling exit() ?

File: src/com/company/Main.java [hunk 11]
```diff
@@ -8,18 +8,40 @@
...
-        searchtheword(data,searchword);
+
+        String data = readFileAsString(txtFilePath);
+
+
+        String searchword = null;
+        try {
+            searchword = args[1];
+        } catch (Exception e) {
+            System.out.println("search word has not entered");
```
Comment (premsrj) [inline src/com/company/Main.java]:
```suggestion
            System.out.println("word to search was not entered");
```

File: src/com/company/Main.java [hunk 12]
```diff
@@ -33,43 +55,34 @@ public static String readFileAsString(String fileName)
...
-                System.out.println("file exists");
-            }
-            else {
-                System.out.println("file doesn't found ");
-            }
-        }
-        public static void avoidspecialchar(String data) {
-            data = data.replaceAll("[^a-zA-Z0-9@]", " ");
+        public static boolean isFileExists(File file) {
+           return file.exists();
```
Comment (premsrj) [inline src/com/company/Main.java]: Why have a different method if it is just a one line check? You can include file.exists() directly in the if condition

File: src/com/company/Main.java [hunk 13]
```diff
@@ -8,18 +8,40 @@
 
 
 public class Main {
-    public static void main(String[] args){
+    public static void main(String[] args) {
+
+        String txtFilePath = args[0];
+
+        File file = new File(txtFilePath);
```
Comment (premsrj) [inline src/com/company/Main.java:23]: What if the user did not provide any arguments?

File: src/com/company/Main.java [hunk 14]
```diff
@@ -8,18 +8,40 @@
...
+            System.out.println("Enter the valid path ");
+            System.exit(0);
+        }
+        isFileFormatSupported(file);
+        isFileExists(file);
+
+        if((isFileExists(file) && isFileFormatSupported(file))== false)
+        {
+            System.out.println("file is not valid");
+            System.exit(0);
```
Comment (premsrj) [inline src/com/company/Main.java]: Why not just use return statement here?

File: src/com/company/Main.java [hunk 15]
```diff
@@ -8,18 +8,40 @@
...
+        File file = new File(txtFilePath);
+        if(file.length()==0)
+        {
+            System.out.println("Enter the valid path ");
+            System.exit(0);
+        }
+        isFileFormatSupported(file);
+        isFileExists(file);
+
+        if((isFileExists(file) && isFileFormatSupported(file))== false)
```
Comment (premsrj) [inline src/com/company/Main.java]:
```suggestion
        if((isFileExists(file) && isFileFormatSupported(file)) == false) {
```
Proper spacing for == and { in the same line

File: src/com/company/Main.java [hunk 16]
```diff
@@ -33,43 +55,34 @@ public static String readFileAsString(String fileName)
...
         }
-            public static void searchtheword(String data,String searchword) {
+
+            public static void searchTheWord(String data,String searchword) {
                 StringTokenizer st = new StringTokenizer(data);
                 int count = 0;
+                if(st.hasMoreElements()==false)
+                {
+                    System.out.println("file is empty");
+                    System.exit(0);
```
Comment (premsrj) [inline src/com/company/Main.java]: return statement?

File: src/com/company/Main.java [hunk 17]
```diff
@@ -8,18 +8,40 @@
 
 
 public class Main {
-    public static void main(String[] args){
+    public static void main(String[] args) {
+
+        String txtFilePath = args[0];
+
+        File file = new File(txtFilePath);
+        if(file.length()==0)
```
Comment (premsrj) [inline src/com/company/Main.java]:
```suggestion
        if(file.length()==0) {
```
curly bracket should be in the same line

Comment (kssreejith1994) [inline src/com/company/Main.java]: What does file.length() do? Does it validates the file path? The error message inside if statement says "enter valid path". Please check.

File: src/com/company/Main.java [hunk 18]
```diff
@@ -8,18 +8,40 @@
 
 
 public class Main {
-    public static void main(String[] args){
+    public static void main(String[] args) {
+
+        String txtFilePath = args[0];
```
Comment (kssreejith1994) [inline src/com/company/Main.java:21]: ArrayIndexOutOfBounds exception will be thrown here if user does not pass any argument. Please check.

**Review by premsrj — CHANGES_REQUESTED** [review]

File: src/com/company/Main.java [hunk 19]
```diff
@@ -8,18 +8,49 @@
...
 public class Main {
-    public static void main(String[] args){
+    public static void main(String[] args) {
+
+
+        String txtFilePath = null;
+        try {
+            txtFilePath = args[0];
+        } catch (Exception e) {
+            System.out.println("Path has not entered");
```
Comment (premsrj) [inline src/com/company/Main.java]:
```suggestion
            System.out.println("Path was not entered");
```

File: src/com/company/Main.java [hunk 20]
```diff
@@ -8,18 +8,49 @@
...
+            System.exit(1);
```
Comment (premsrj) [inline src/com/company/Main.java]: Return statement instead of exit() ?

**Review by premsrj — CHANGES_REQUESTED** [review]

File: src/com/company/Main.java (hunk 15 above)
Comment (kssreejith1994) [inline src/com/company/Main.java]: You don't have to check against false again, since isFileExists and isFileFormat themselves return boolean. Please check.

File: src/com/company/Main.java [hunk 21]
```diff
@@ -8,18 +8,40 @@
...
+        {
+            System.out.println("Enter the valid path ");
+            System.exit(0);
+        }
+        isFileFormatSupported(file);
+        isFileExists(file);
+
+        if((isFileExists(file) && isFileFormatSupported(file))== false)
+        {
+            System.out.println("file is not valid");
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Code is duplicated here. Can we convert this to a method that takes a message as argument and prints it and exits? This way you can call the method when you want the program to exit. Check line 18 and 19 to relate this.

File: src/com/company/Main.java [hunk 22]
```diff
@@ -8,18 +8,40 @@
...
-        data = readFileAsString(txtfile);
-        avoidSpecialChar(data);
-        searchTheWord(data,searchword);
+
+        String data = readFileAsString(txtFilePath);
+
+
+        String searchword = null;
+        try {
+            searchword = args[1];
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Check this at the starting of the program. File path and search word are mandatory for you program to execute. So if they are not present, then you don't need to continue further.

File: src/com/company/Main.java [hunk 23]
```diff
@@ -33,44 +55,34 @@ public static String readFileAsString(String fileName)
               } catch (IOException e) {
                   e.printStackTrace();
               }
-              return data;
+              return data.replaceAll("[^a-zA-Z0-9@]", " ");
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: The regex pattern can be made final/constant

File: src/com/company/Main.java [hunk 24]
```diff
@@ -33,44 +55,34 @@ public static String readFileAsString(String fileName)
...
-            if (file.getName().endsWith(".txt")) {
-                System.out.println("file supported");
-
-            }
-
-            else
-                {
-                    System.out.println("file not supported");
-                }
+            return (file.getName().endsWith(".txt"));
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Make file extension also final/constant

File: src/com/company/Main.java [hunk 25]
```diff
@@ -33,44 +55,34 @@ public static String readFileAsString(String fileName)
...
-        {
-            data = data.replaceAll("[^a-zA-Z0-9@]", " ");
+        public static boolean isFileExists(File file) {
+           return file.exists();
         }
+
             public static void searchTheWord(String data,String searchword) {
                 StringTokenizer st = new StringTokenizer(data);
                 int count = 0;
+                if(st.hasMoreElements()==false)
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: check what does st.hasMoreElements() do

**Review by kssreejith1994 — CHANGES_REQUESTED** [review]
Please make the requested changes.

**Comment by kssreejith1994** [comment]
Also, please add a relatable commit message everytime.

**Comment by premsrj** [comment]
> Also, please add a relatable commit message everytime.

I agree, all the commits have the same message. Your commit message should have information of what was actually done in that commit.

File: src/com/company/Main.java [hunk 26]
```diff
@@ -8,68 +8,53 @@
...
 public class Main {
-    public static void main(String[] args){
+    public static void main(String[] args) {
 
-        String txtfile = args[0];
-        File file = new File(txtfile);
-        checkforsupport(file);
-        checkforexistance(file);
+        if(args.length!=2) {
+            System.out.println("Please enter the valid path and word");
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: May be you can rephrase this to "Exiting, since no file path and word to search for are provided!"

File: src/com/company/Main.java (hunk 1 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: i have changed the code with dispalying the message if user does not passes any one output

File: src/com/company/Main.java (hunk 2 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: i have solved this with checking the args array and continue if required args are given else the execution is stopped

File: src/com/company/Main.java (hunk 3 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: if file is empty the message is passed like the file has no data and execution will be terminated

File: src/com/company/Main.java (hunk 25 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: it was an extra code and i have removed that

File: src/com/company/Main.java (hunk 24 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: i have made the change

File: src/com/company/Main.java (hunk 22 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: yeah, I have made that with the if statement.

File: src/com/company/Main.java (hunk 4 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: i have  used return statement to terminate the execution if it prints "file does not exists" statement

File: src/com/company/Main.java (hunk 6 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: i have changed the method naming format for all the methods.

File: src/com/company/Main.java (hunk 7 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: i have resolved that in a  recently committed code.

File: src/com/company/Main.java (hunk 8 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: i have made to print as "file is not valid" if it is not in a supported format.

File: src/com/company/Main.java (hunk 9 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: yeah i have made this change in  recently updated code.

File: src/com/company/Main.java (hunk 10 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: i have replaced all the exit() statements with the return statements.

File: src/com/company/Main.java (hunk 11 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: done with that change.

File: src/com/company/Main.java (hunk 13 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java:23]: it prints the the error message as "Please enter the valid path and word" and terminate the execution.

File: src/com/company/Main.java (hunk 15 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: I have done that.
ill check for other lines of code if there is any other issue repeated as this.

File: src/com/company/Main.java (hunk 20 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: I have replaced it with return statement.

File: src/com/company/Main.java (hunk 5 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: I have changed all the method names.

File: src/com/company/Main.java (hunk 12 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: we can do that and I have done that in recently pushed code.

File: src/com/company/Main.java (hunk 16 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: changed exit() to return statement.

File: src/com/company/Main.java (hunk 17 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: done that for all the lines which has this scenarios.

File: src/com/company/Main.java (hunk 19 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: changed with the println statement.

File: src/com/company/Main.java (hunk 17 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: changed the print statement to  "the file does not contains any data".

File: src/com/company/Main.java (hunk 18 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java:21]: resolved that exception and updated in the recently committed code

File: src/com/company/Main.java (hunk 15 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: I have changed the statement.

File: src/com/company/Main.java (hunk 23 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: done with that.

**Comment by premsrj** [comment]
@vikasreddy-elixr Now you have three commits with the message 
"I have replaced all the exit() with return statement. "
Is that all that was done in the three commits?

File: src/com/company/Main.java [hunk 27]
```diff
@@ -8,68 +8,53 @@
 
 
 public class Main {
```
Comment (premsrj) [inline src/com/company/Main.java:9]: Code seems ok now. Fix the file formatting. The indentation and spaces are all wrong. There is a menu option available in IntelliJ

Code -> Reformat File

**Review by premsrj — CHANGES_REQUESTED** [review]

File: src/com/company/Main.java (hunk 14 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: replaced exit() with the return statement

File: src/com/company/Main.java (hunk 27 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java:9]: done with File formatting

File: src/com/company/Main.java [hunk 28]
```diff
@@ -20,53 +21,55 @@ public static void main(String[] args) {
...
+        if (count == 0) {
+            System.out.println("Word not found");
+        } else {
+            System.out.println("The word has been found");
+            System.out.println("the word has been repeated for " + count + " times");
+        }
+    }
+
+}
+
```
Comment (premsrj) [inline src/com/company/Main.java]: Can you remove these extra lines from 73 to 77?

**Review by premsrj — CHANGES_REQUESTED** [review]

File: src/com/company/Main.java [hunk 29]
```diff
@@ -20,53 +21,55 @@ public static void main(String[] args) {
 
         isFileFormatSupported(file);
 
-        if(!file.exists() || !isFileFormatSupported(file)) {
+        if (!file.exists() || !isFileFormatSupported(file)) {
```
Comment (kssreejith1994) [inline src/com/company/Main.java:25]: isFileFormatSupported is called in line 22 as well?

File: src/com/company/Main.java [hunk 30]
```diff
@@ -20,53 +21,55 @@ public static void main(String[] args) {
...
             System.out.println("file is not valid");
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Start all the sysout messages with capital letters and end with proper punctuations.

File: src/com/company/Main.java (hunk 28 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: yeah sure
I have removed it

File: src/com/company/Main.java [hunk 31]
```diff
@@ -20,53 +21,55 @@ public static void main(String[] args) {
...
-        public static void searchTheWord(String data,String searchword) {
-                StringTokenizer st = new StringTokenizer(data);
-                int count = 0;
+        return data.replaceAll("[^a-zA-Z0-9@]", " ");
+    }
 
-                while (st.hasMoreTokens())
-                {
+    public static boolean isFileFormatSupported(File file) {
+        return (file.getName().endsWith(".txt"));
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Instead of hardcoding .txt, you could probably make this is a final variable?

File: src/com/company/Main.java [hunk 32]
```diff
@@ -20,53 +21,55 @@ public static void main(String[] args) {
...
+        String data = "";
+        try {
+            data = new String(Files.readAllBytes(Paths.get(fileName)));
+        } catch (IOException e) {
+            e.printStackTrace();
         }
-        public static void searchTheWord(String data,String searchword) {
-                StringTokenizer st = new StringTokenizer(data);
-                int count = 0;
+        return data.replaceAll("[^a-zA-Z0-9@]", " ");
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Instead of hardcoding regex here, you could probably make this is a final variable?

**Review by kssreejith1994 — CHANGES_REQUESTED** [review]
Code looks okay, few minor comments to check.

File: src/com/company/Main.java (hunk 26 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: yeah changes done.

File: src/com/company/Main.java (hunk 29 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java:25]: it was duplicated,
but now the duplicated code has been removed.

File: src/com/company/Main.java (hunk 30 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: done with that.

File: src/com/company/Main.java (hunk 31 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: created the variable and made that as you suggested

File: src/com/company/Main.java (hunk 32 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: created the variable and made that as you suggested.

File: src/com/company/Main.java [hunk 33]
```diff
@@ -8,82 +8,60 @@
...
 
 public class Main {
-    public static void main(String[] args){
-
-        String txtfile = args[0];
-        File file = new File(txtfile);
-        checkforsupport(file);
-        checkforexistance(file);
+    final static String extention=".txt";
+    final static String specialChars="[^a-zA-Z0-9@]";
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: constants should be defined in capitals and should be meaningful. specialChars does not serve the purpose of the regex used.

**Review by kssreejith1994 — CHANGES_REQUESTED** [review]

File: src/com/company/Main.java [hunk 34]
```diff
@@ -8,82 +8,60 @@
...
-              }
-              return data;
+    public static String readFileAsString(String fileName) {
+        String data =null;
+        try {
+            data = new String(Files.readAllBytes(Paths.get(fileName)));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return data.replaceAll(SPECIAL_CHAR_REMOVAL_REGEX, SINGLE_SPACE);
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: There is a possibility of null pointer exception here. What happens if there is no data in file and program execution continues?

File: src/com/company/Main.java [hunk 35]
```diff
@@ -8,82 +8,60 @@
...
-
-        }
-
-
-
-
-
-
+    }
+}
```
Comment (kssreejith1994) [inline src/com/company/Main.java:75]: Add a new line here.

**Review by kssreejith1994 — CHANGES_REQUESTED** [review]

File: src/com/company/Main.java [hunk 36]
```diff
@@ -4,86 +4,67 @@
...
 
 
 public class Main {
-    public static void main(String[] args){
-
-        String txtfile = args[0];
-        File file = new File(txtfile);
-        checkforsupport(file);
-        checkforexistance(file);
+    final static String FILE_TYPE_1=".txt";
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Make it FILE_TYPE_TXT please

File: src/com/company/Main.java [hunk 37]
```diff
@@ -4,86 +4,67 @@
...
+    final static String FILE_TYPE_2=".jason";
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Make it FILE_TYPE_JSON please. and its json.

File: src/com/company/Main.java [hunk 38]
```diff
@@ -4,86 +4,67 @@
...
+        String data =null;
+        try {
+            data = new String(Files.readAllBytes(Paths.get(fileName)));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if(data!=null) {
+            return data.replaceAll(SPECIAL_CHAR_REMOVAL_REGEX, SINGLE_SPACE);
+        }
+        return data;
```
Comment (kssreejith1994) [inline src/com/company/Main.java:52]: Data will still be null. Please do the null check in main.

**Review by kssreejith1994 — CHANGES_REQUESTED** [review]

File: src/com/company/Main.java [hunk 39]
```diff
@@ -1,89 +1,72 @@
...
+            return;
+        }
+        if (file.length() == 0) {
+            System.out.println("The file does not contains any data ");
+            return;
+        }
+        System.out.println("Processing................");
+        String data = readFileAsString(txtFilePath);
+        if (data == null) {
+            return;
```
Comment (kssreejith1994) [inline src/com/company/Main.java:38]: Add a sysout mentioning that since no data present in file,program is exiting

**Review by kssreejith1994 — CHANGES_REQUESTED** [review]

File: src/com/company/Main.java [hunk 40]
```diff
@@ -1,89 +1,70 @@
...
-                   data = new String(Files.readAllBytes(Paths.get(fileName)));
-              } catch (IOException e) {
-                  e.printStackTrace();
-              }
-              return data;
+    public static String readFileAsString(String fileName) {
+        String data = null;
+        try {
+            data = new String(Files.readAllBytes(Paths.get(fileName)));
+            data.replaceAll(SPECIAL_CHAR_REMOVAL_REGEX, SINGLE_SPACE);
```
Comment (kssreejith1994) [inline src/com/company/Main.java]: Shouldn't it be data=data.replaceAll() ? Before changing to this try to read a file that contains ',' in the text and see the output.

**Review by kssreejith1994 — CHANGES_REQUESTED** [review]

File: src/com/company/Main.java (hunk 33 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: done with the changes

File: src/com/company/Main.java (hunk 34 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: resolved with writing replaceall method within try block

File: src/com/company/Main.java (hunk 40 above)
Comment (vikasreddy-elixr) [reply src/com/company/Main.java]: yeah checked that
data replaced was not storing in the variable data

**Review by premsrj — APPROVED** [review]

**Review by kssreejith1994 — APPROVED** [review]
Code looks fine now.

**Comment by kssreejith1994** [comment]
Merged code to master.
